@xen-orchestra/xapi/src/vdi.js

```diff
@@ -7,6 +7,16 @@ module.exports = class Vdi {
     return extractOpaqueRef(await this.callAsync('VDI.clone', vdiRef))
   }
 
+  async snapshot(vdiRef, nameLabel) {
+    const snap = await this._getOrWaitObject(extractOpaqueRef(await this.callAsync('VDI.snapshot', vdiRef)))
+
+    if (nameLabel) {
+      await snap.set_name_label(nameLabel)
+    }
+
+    return snap
+  }
```

Comment:
Methods in this lib should be as light as possible, they should take a ref and return a ref :slightly_smiling_face:

Reply:
Also, they should rely as little as possible on events and cached objects.

If possible, prefer fetching specific properties.